### DISCUSSION TITLE
Undefined `stylesheet_pack_tag` when upgrading from Webpacker to Shakapacker 6

### DISCUSSION THREAD

**Discussion by ThePeach** [shakacode/shakapacker#510] [Q&A] [answered]
This is a long shot, but I'm going to try anyway: I'm currently upgrading webpacker into shakapacker as this was long due. 
The project uses some JS and recently switched over to React.
I have followed the upgrade guides as suggested from 4 to 5, from 5 to 6.0.0-rc6 and then from webpacker to shakapacker, and then upgrading to Shakapacker 6.5.2.
Now I think I got the majority of the configuration sorted but for some reason now I'm getting an error when trying to load the main page which I cannot understand why.

The full error is
```
undefined method `stylesheet_pack_tag' for #<ActionView::Base:0x0000000002e298>
Did you mean?  stylesheet_path
               stylesheet_link_tag
```

On top of that, and I don't know if that's intended, I don't have the rails task `assets:precompile` anymore... I feel something around the gems installation has gone awry... Any idea what could have caused this? Consider that I didn't touch the gems aside from webpacker/shakapacker's.

I did cleanup all gems with `gem uninstall -aIx` and then re-issued `bundler install` just to make sure there wasn't anything wrong, and I got no error from that.

I am wondering if moving forward with the upgrade to the latest version of Shakapacker might be a good idea, or it's better to stop here and try to solve this problem first.

**Comment by tomdracz** [comment]
Weird one. Missing method name does seem to indicate that something is not loaded. Missing `assets:precompile` does make me wonder if the problem is elsewhere.

When the gem is loaded, it should effectively load this file https://github.com/shakacode/shakapacker/blob/main/lib/shakapacker.rb#L49 and at the end load railtie that pulls in relevant view helpers et al. Wonder if this is not running because Rails is not defined for some reason?

Webpacker worked in exact same way so I would see if there were some other gem upgrades along the way and maybe work backwards from that.

I would maybe try and fire up Rails console and see if `Shakapacker` is even defined. If it is, then see if `Shakapacker::Helper` is defined.

If the `Shakapacker` is missing, then there's something very wrong with the gem load - just not loading anything. If it's there but the `Shakapacker::Helper` isn't, then that would indicate Railtie didn't load and `Rails` is not defined (???)

**Reply by ThePeach** [reply]
Thanks for the comment @tomdracz, I've re-run the `webpacker:install` and went through it again just in case and it seems like that got me away from it, only now I'm stuck with an error from `webpack-cli` that suggests some error in the configuration options which I have no idea what could be the cause of it, I feel this is due to some mismatch of webpack versions:

```
▶ bin/webpacker-dev-server
Loading ENV specific webpack configuration file /Users/peach/repos/faircare/config/webpack/development.js
[webpack-cli] Invalid options object. Dev Server has been initialized using an options object that does not match the API schema.
 - options has an unknown property '_assetEmittingPreviousFiles'. These properties are valid:
   object { allowedHosts?, bonjour?, client?, compress?, devMiddleware?, headers?, historyApiFallback?, host?, hot?, ipc?, liveReload?, onListening?, open?, port?, proxy?, server?, setupExitSignals?, setupMiddlewares?, static?, watchFiles?, webSocketServer? }
```

in `package.json` shakapacker added the following:
```
    "webpack": "5",
    "webpack-assets-manifest": "5",
    "webpack-cli": "4",
    "webpack-dev-server": "^5.0.4",
    "webpack-merge": "5",
```

the installed version of webpack is 5.94.0 and webpack-cli is 4.10.0

do you think it'll be fine upgrading webpack-cli to v5?

**Comment by ThePeach** [comment, marked as answer]
ok found that I missed some warnings in the output of yarn which forced me to upgrade some packages, this solved the problem, sorry for the bother.
